packages/@uppy/transloadit/src/Client.js

```diff
@@ -12,7 +11,22 @@ function fetchJSON (...args) {
     }
 
     if (!response.ok) {
-      return Promise.reject(new NetworkError(response.statusText))
+      const serverError = new Error(response.statusText)
+      return response.json().then(assembly => {
+        if (!assembly.error) throw serverError
+
+        const error = new Error(assembly.error)
+        error.details = assembly.message
+        error.assembly = assembly
+        if (assembly.assembly_id) {
+          error.details += ` Assembly ID: ${assembly.assembly_id}`
+        }
+        throw error
```

Comment:
this stuff is quite specific to assembly creation I think 🤔 it probably works fine like this but if it could be done only for `createAssembly()` calls it would be even better

Reply:
I couldn't find a "good" way to special case the assembly creation request (I don't want to create a function just for it, I don't want to leak the response object to the error object, I don't think it's worth using `WeakMap`s), so I end up using `url.endsWith('/assemblies')` which may have some false positive. Open to better suggestions :)

Reply:
I couldn't find a "good" way to special case the assembly creation request (I don't want to create a function just for it, I don't want to leak the response object to the error object, I don't think it's worth using `WeakMap`s), so I end up using `url.endsWith('/assemblies')` which may have some false positive. Open to better suggestions :)

Reply:
Right, I was thinking of putting the response object on the error object, but if you dont want to do that I'm OK with this solution :+1:

Reply:
I’ve read somewhere that it was bad practice as it could create memory leaks to have `Response` objects on the error.